src/main/kotlin/network/warzone/mars/utils/ClosedPolymorphismDeserializer.kt

```diff
@@ -0,0 +1,67 @@
+package network.warzone.mars.utils
+
+import com.google.gson.*
+import kotlinx.serialization.SerialName
+import network.warzone.mars.api.socket.models.PlayerUpdateData
+import kotlin.reflect.KClass
+import kotlin.reflect.full.findAnnotation
+
+object ClosedPolymorphismDeserializer {
```

Comment:
This is just me brainstorming, but wondering if there's a better way to do this. If we had a consistent deserializer for both the API and server I don't think this would be necessary but that is more of a future task if anything (and potentially irrelevant if we switch to the new API implementation).

Reply:
In the future I think it makes sense to:
* Switch to `kotlinx.serialization` for [native sealed class support](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization/-sealed-class-serializer/)
* If we switch over to Rust API, writing a custom derive macro for the Serialize trait which can understand the `kotlinx.serialization` polymorphic discriminator, which we can simply add to polymorphic models